src/main/java/se/sundsvall/contract/api/model/Contract.java

```diff
@@ -1,49 +1,55 @@
 package se.sundsvall.contract.api.model;
 
+import java.time.LocalDate;
 import java.util.List;
 
-import com.fasterxml.jackson.annotation.JsonSubTypes;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import org.geojson.FeatureCollection;
 
 import se.sundsvall.contract.model.ExtraParameterGroup;
+import se.sundsvall.contract.model.Fees;
 import se.sundsvall.contract.model.TermGroup;
 import se.sundsvall.dept44.common.validators.annotation.OneOf;
 
 import io.swagger.v3.oas.annotations.media.ArraySchema;
 import io.swagger.v3.oas.annotations.media.Schema;
 import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
 import lombok.NoArgsConstructor;
-import lombok.experimental.SuperBuilder;
-
-@JsonTypeInfo(
-	use = JsonTypeInfo.Id.NAME,
-	include = JsonTypeInfo.As.EXISTING_PROPERTY,
-	property = "type"
-)
-@JsonSubTypes({
-	@JsonSubTypes.Type(value = LandLeaseContract.class, name = "LAND_LEASE")
-})
+
 @Data
-@EqualsAndHashCode
-@SuperBuilder(setterPrefix = "with")
+@Builder(setterPrefix = "with")
+@AllArgsConstructor
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
 @Schema(description = "Contract")
-public abstract class Contract {
+public class Contract {
 
-	/*
-	 * Backed by enum {@link se.sundsvall.contract.api.model.enums.ContractType}
-	 */
-	@OneOf(value = {"LAND_LEASE"}, nullable = true)
-	@Schema(description = "Contract type. Possible values: LAND_LEASE", example = "LAND_LEASE")
-	private String type;
+	@Schema(description = "Version for contract", example = "1", accessMode = Schema.AccessMode.READ_ONLY)
+	private int version;
 
 	@Schema(description = "Contract id", example = "2024-12345", accessMode = Schema.AccessMode.READ_ONLY)
 	private String contractId;
 
-	@Schema(description = "Version for contract", example = "1", accessMode = Schema.AccessMode.READ_ONLY)
-	private int version;
+	@Schema(description = "A description ", example = "A simple description of the contract")
+	private String description;
+
+	@Schema(description = "External referenceId", example = "123")
+	private String externalReferenceId;
+
+	/*
+	 * Backed by enum {@link se.sundsvall.contract.api.model.enums.LandLeaseType}
+	 */
+	@Schema(description = "Type of lease", example = "LEASEHOLD", nullable = true)
+	@OneOf({"LEASEHOLD", "USUFRUCT", "SITELEASEHOLD"})
```

Comment:
Optional/nullable, since it relates to land-leases and maybe not to other contract types?

Reply:
I think the nullable should be on the `@OneOf`, right?

Reply:
Yes